goblet/tests/test_pubsub.py

```diff
@@ -1,502 +1,108 @@
-        monkeypatch.setenv("GOOGLE_PROJECT", "goblet")
-        monkeypatch.setenv("GOOGLE_LOCATION", "us-central1")
-        monkeypatch.setenv("G_TEST_NAME", "pubsub-deploy-cloudrun")
+        monkeypatch.setenv("G_TEST_NAME", "pubsub-deploy")
         monkeypatch.setenv("G_HTTP_TEST", "REPLAY")
 
-        cloudrun_url = "https://goblet-12345.a.run.app"
-        service_account = "SERVICE_ACCOUNT@developer.gserviceaccount.com"
-
-        pubsub = PubSub(
-            "goblet",
-            backend=CloudRun(
-                Goblet(
-                    backend="cloudrun",
-                    config={"pubsub": {"serviceAccountEmail": service_account}},
-                )
-            ),
-        )
-        pubsub.register("test", None, kwargs={"topic": "test", "kwargs": {}})
-
-        pubsub._deploy()
+        app = Goblet(function_name="goblet_example")
 
-        responses = get_responses("pubsub-deploy-cloudrun")
+        client: PubSubClient = app.pubsub_topic(name="test")  # noqa: F841
 
-        assert len(responses) == 3
-        assert responses[1]["body"]["status"]["url"] == cloudrun_url
-        assert (
-            responses[2]["body"]["pushConfig"]["oidcToken"]["serviceAccountEmail"]
-            == service_account
-        )
-        assert (
-            responses[2]["body"]["pushConfig"]["oidcToken"]["audience"] == cloudrun_url
+        app.destroy(
+            skip_backend=True,
         )
-        assert responses[2]["body"]["pushConfig"]["pushEndpoint"] == cloudrun_url
-        assert responses[2]["body"]["topic"] == "projects/goblet/topics/test"
 
-    def test_destroy_pubsub_cloudrun(self, monkeypatch):
-        monkeypatch.setenv("GOOGLE_PROJECT", "goblet")
-        monkeypatch.setenv("GOOGLE_LOCATION", "us-central1")
-        monkeypatch.setenv("G_TEST_NAME", "pubsub-destroy-cloudrun")
-        monkeypatch.setenv("G_HTTP_TEST", "REPLAY")
-
-        pubsub = PubSub(
-            "goblet",
-            resources={"test": {"trigger": {}, "subscription": {"test": {}}}},
-            backend=CloudRun(Goblet(backend="cloudrun")),
+        delete_pubsub_topic = get_response(
+            "pubsub-deploy",
+            "delete-v1-projects-goblet-topics-test_1.json",
         )
-        pubsub.destroy()
 
-        responses = get_responses("pubsub-destroy-cloudrun")
+        pubsub_topic = app.infrastructure["pubsub_topic"]
 
-        assert len(responses) == 1
-        assert responses[0]["body"] == {}
+        assert delete_pubsub_topic["body"] == {}
+        assert pubsub_topic.resource["test"]["id"] == "test"
 
-    def test_update_pubsub_subscription(self, monkeypatch):
+    def test_update_pubsub_topic(self, monkeypatch):
         monkeypatch.setenv("GOOGLE_PROJECT", "goblet")
         monkeypatch.setenv("GOOGLE_LOCATION", "us-central1")
-        monkeypatch.setenv("G_TEST_NAME", "pubsub-update-subscription")
+        monkeypatch.setenv("G_TEST_NAME", "pubsub-deploy")
```

Comment:
this should be a seperate folder for this test. `pubsub-update`. we can just copy the files we need over to that new folder.

Reply:
done!